openmc/mgxs/mgxs.py

```diff
@@ -5011,13 +5028,1153 @@ class NuFissionMatrixXS(MatrixMGXS):
+    sparse : bool
+        Whether or not the MGXS' tallies use SciPy's LIL sparse matrix format
+        for compressed data storage
+    loaded_sp : bool
+        Whether or not a statepoint file has been loaded with tally data
+    derived : bool
+        Whether or not the MGXS is merged from one or more other MGXS
+    hdf5_key : str
+        The key used to index multi-group cross sections in an HDF5 data store
+    mgxs : list of openmc.mgxs.MGXS
+        A list of MGXS to combine to compute this multi-group cross section
+
+    """
+
+    def __init__(self, domain=None, domain_type=None, groups=None,
+                 by_nuclide=False, name='', num_polar=1, num_azimuthal=1):
+
+        self._mgxs = []
+        super(ConvolvedMGXS, self).__init__(
+            domain, domain_type, groups, by_nuclide,
+            name, num_polar, num_azimuthal)
+
+    def __deepcopy__(self, memo):
+        clone = super(ConvolvedMGXS, self).__deepcopy__(memo)
+        clone._mgxs = copy.deepcopy(self.mgxs)
+        return clone
+
+    @property
+    def mgxs(self):
+        return self._mgxs
+
+    @property
+    def scores(self):
+        scores = []
+        for mgxs in self.mgxs:
+            scores += mgxs.scores
+        return scores
+
+    @property
+    def filters(self):
+        filters = []
+        for mgxs in self.mgxs:
+            filters.extend(mgxs.filters)
+        return filters
+
+    @property
+    def tally_keys(self):
+        keys = []
+        for mgxs in self.mgxs:
+            for tally_key in mgxs.tallies:
+                keys += ['{} : {}'.format(mgxs.hdf5_key, tally_key)]
+        return keys
+
+    @property
+    def tallies(self):
+
+        # Instantiate tallies if they do not exist
+        if self._tallies is None:
+
+            # Initialize a collection of Tallies
+            self._tallies = OrderedDict()
+
+            # Add tallies created for each MGXS
+            for mgxs in self.mgxs:
+                for tally_key in mgxs.tallies:
+                    new_key = '{} : {}'.format(mgxs.hdf5_key, tally_key)
+                    self._tallies[new_key] = mgxs.tallies[tally_key]
+
+        return self._tallies
+
+    @MGXS.name.setter
+    def name(self, name):
+        cv.check_type('name', name, string_types)
+
+        self._name = name
+        for mgxs in self.mgxs:
+            mgxs.name = name
+
+    @MGXS.by_nuclide.setter
+    def by_nuclide(self, by_nuclide):
+        cv.check_type('by_nuclide', by_nuclide, bool)
+
+        self._by_nuclide = by_nuclide
+        for mgxs in self.mgxs:
+            mgxs.by_nuclide = by_nuclide
+
+    @MGXS.nuclides.setter
+    def nuclides(self, nuclides):
+        cv.check_iterable_type('nuclides', nuclides, string_types)
+
+        self._nuclides = nuclides
+        for mgxs in self.mgxs:
+            mgxs.nuclides = nuclides
+
+    @MGXS.estimator.setter
+    def estimator(self, estimator):
+        raise ValueError('Unable to assign a tally estimator a ConvolvedMGXS')
```

Comment:
Well I didn't think it made sense to allow for this since each of the "sub"-`MGXS` in the convolution may accept different estimators. This is indeed the case for both subclasses of `ConvolvedMGXS` implemented here - `ScatterProbabilityMatrixXS` is a convolution of `ScatterXS` and `ScatterProbabilityMatrixXS` which support analog or tracklength and analog only estimators, respectively. Would you prefer that `ConvolvedMGXS` attempt to assign the tally estimators to each "sub"-`MGXS` to occur at their level instead?